csharp/test/Apache.Arrow.Adbc.Tests/Metadata/GetObjectsParser.cs

```diff
@@ -161,26 +161,34 @@ private static List<AdbcConstraint> ParseConstraints(StructArray constraintsArra
 
             StringArray name = (StringArray)constraintsArray.Fields[StandardSchemas.ConstraintSchema.FindIndex(f => f.Name == "constraint_name")]; // constraint_name | utf8
             StringArray type = (StringArray)constraintsArray.Fields[StandardSchemas.ConstraintSchema.FindIndex(f => f.Name == "constraint_type")]; //	constraint_type | utf8 not null
-            ListArray column_names = (ListArray)constraintsArray.Fields[StandardSchemas.ConstraintSchema.FindIndex(f => f.Name == "constraint_column_names")]; //	constraint_column_names | list<utf8> not null
-            ListArray column_usage = (ListArray)constraintsArray.Fields[StandardSchemas.ConstraintSchema.FindIndex(f => f.Name == "constraint_column_usage")]; //	constraint_column_usage | list<USAGE_SCHEMA>
+            ListArray columnNames = (ListArray)constraintsArray.Fields[StandardSchemas.ConstraintSchema.FindIndex(f => f.Name == "constraint_column_names")]; //	constraint_column_names | list<utf8> not null
+            ListArray columnUsages = (ListArray)constraintsArray.Fields[StandardSchemas.ConstraintSchema.FindIndex(f => f.Name == "constraint_column_usage")]; //	constraint_column_usage | list<USAGE_SCHEMA>
 
             for (int i = 0; i < constraintsArray.Length; i++)
             {
                 AdbcConstraint c = new AdbcConstraint();
                 c.Name = name.GetString(i);
                 c.Type = type.GetString(i);
 
-                StringArray col_names = column_names.GetSlicedValues(i) as StringArray;
-                StructArray usage = column_usage.GetSlicedValues(i) as StructArray;
+                StringArray colNames = columnNames.GetSlicedValues(i) as StringArray;
+                StructArray usages = columnUsages.GetSlicedValues(i) as StructArray;
 
-                if (usage != null)
+                if (colNames != null)
                 {
-                    for (int j = 0; j < usage.Length; j++)
+                    for (int j = 0; j < colNames.Length; j++)
                     {
-                        StringArray fkCatalog = (StringArray)usage.Fields[StandardSchemas.UsageSchema.FindIndex(f => f.Name == "fk_catalog")]; // fk_catalog	| utf8
-                        StringArray fkDbSchema = (StringArray)usage.Fields[StandardSchemas.UsageSchema.FindIndex(f => f.Name == "fk_db_schema")]; //fk_db_schema | utf8
-                        StringArray fkTable = (StringArray)usage.Fields[StandardSchemas.UsageSchema.FindIndex(f => f.Name == "fk_table")]; //	fk_table | utf8 not null
-                        StringArray fkColumnName = (StringArray)usage.Fields[StandardSchemas.UsageSchema.FindIndex(f => f.Name == "fk_column_name")]; // fk_column_name | utf8 not null
+                        c.ColumnNames.Add(colNames.GetString(j));
+                    }
+                }
+
+                if (usages != null)
+                {
+                    for (int j = 0; j < usages.Length; j++)
```

Comment:
This for loop looks like it should start after all these Find() calls since they don't depend on j -- we are redundantly running them on each iteration. Additionally we should add an assertion that FindIndex() never returns -1.

Reply:
Let's examine if this can be applied elsewhere.

Reply:
Added extension method to perform additional error handling and throw exception if the index returned is -1.